moveit_simple/src/moveit_simple.cpp

```diff
@@ -770,6 +770,21 @@ void Robot::updateState(const sensor_msgs::JointStateConstPtr& msg)
   robot_state_->setVariablePositions(msg->name, msg->position);
 }
 
+std::vector<double> Robot::getCurrentRobotState(void) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  ros::spinOnce();
+  std::vector<double> current_joint_positions;
+  current_robot_state_->update();
+  current_robot_state_->copyJointGroupPositions(joint_group_->getName(), current_joint_positions);
+  return current_joint_positions;
+}
+
+
+void Robot::updateCurrentState(const sensor_msgs::JointStateConstPtr& msg)
```

Comment:
Why not stick to a `vector<double>`?  I prefer to not pollute this class with ROS message types except where necessary.